### DISCUSSION TITLE
Is there a way to obtain all referenced "variables" from a compiled template?

### DISCUSSION THREAD

**Discussion by soenkeliebau** [sunng87/handlebars-rust#499] [Q&A]
I have an application that compiles a template once and then renders it multiple times with changing contexts as input.

For my purpose it would be interesting to find out which variable references the template contains, because that would allow me to skip contexts that would result in duplicates.

I have looked at the data structure of a compiled template and it seems like it might be possible to retrieve this, but it looks quite involved. Is there a simple way to obtain this information? Maybe rendering with an empty context and looking at errors that occurred or something similar..

**Comment by sunng87** [comment]
The idea is to find out all `Parameter` in compiled template. You can debug print the compiled template to see its data structure. However, there are cases like nested block, access for upper level of block `../` to resolve. Perhaps an interceptor mechanism is good for this case.
